apps/c_backend/Makefile

```diff
@@ -3,7 +3,7 @@ include ../support/Makefile.inc
 pipeline: pipeline.cpp
 	$(CXX) $(CXXFLAGS) -Wall pipeline.cpp $(LDFLAGS) $(LIB_HALIDE) -o pipeline -lpthread -ldl -lz -g
 
-pipeline_c.c: pipeline
+pipeline_c.cpp: pipeline
```

Comment:
Yes. Examples:

## C++ includes:

```
#include <iostream>
```

## ``extern "C"`` without guard
There's one of these without ``#ifdef __cplusplus``.

```
extern "C" {
...
}
```

## Templates
```
template<typename T> T max(T a, T b) {if (a > b) return a; return b;}
```

## Bool
The ``bool`` type doesn't exist in C but is used in the code emitted by ``compile_to_c()``